### PR TITLE
Add security context to Ray clusters

### PR DESCRIPTION
**Pull request by bjornsen**
Add a default, configurable securityContext to all Ray Cluster pods.

### CODE REVIEW

**Comment by blackzlq** [comment]
I think the latest version is broken, did we test this change on local?

File: modules/kuberay-cluster/variables.tf [hunk 1]
```diff
@@ -57,3 +57,55 @@ variable "gcs_bucket" {
...
+    }))
+    windowsOptions = optional(object({
+      gmsaCredentialSpec    = optional(string)
+      gmsaCredentiaSpecName = optional(string)
+      hostProcess           = bool
+      runAsUserName         = string
+    }))
+  })
+
+  default = {
```
[the remark this answers is not in the document]
Comment (bjornsen) [reply modules/kuberay-cluster/variables.tf:104]: I only took me 2 days but I have finally confirmed this is not a breaking change.

**Pull request merged**